backend/tests/e2e/test_auth_e2e.py

```
@@ -0,0 +1,342 @@
+                            f"Protected endpoint {endpoint} should return {expected_success_code} with valid auth, got {response.status_code}: {response.text}"
+                        print(f"      ✅ Response ({expected_success_code}) - as expected")
+                    else:
+                        # Should fail with specific unauthorized code
+                        assert response.status_code == expected_unauthorized_code, \
+                            f"Protected endpoint {endpoint} should reject {description} with {expected_unauthorized_code}, got {response.status_code}"
+                        print(f"      ✅ Rejected ({response.status_code}) - as expected")
+                else:
+                    # Unprotected endpoints - should always succeed regardless of auth
+                    assert response.status_code == 200, \
+                        f"Unprotected endpoint {endpoint} should work regardless of auth, got {response.status_code}: {response.text}"
+                    print(f"      ✅ Success (200) - as expected")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)